### PR TITLE
add benchmark for fix length input and output

### PR DESCRIPTION
**Pull request by haichuan1221**
vllm only support benchmark for sonnet and sharegpt dataset, but the input and output length is not fixed; in the contribution, I use random sample to evealuate the benchmark of fixed input length and output length

### CODE REVIEW

**Comment by DarkLight1337** [comment]
To speed up the CI queue, I've cancelled the distributed tests for the latest CI run in this PR since they won't pass anyway until #5905 has been merged. Now that it has been merged, please merge `main` into your branch so that the CI can pass once again.

**Comment by haichuan1221** [comment]
> To speed up the CI queue, I've cancelled the distributed tests for the latest CI run in this PR since they won't pass anyway until #5905 has been merged. Now that it has been merged, please merge `main` into your branch so that the CI can pass once again.

I have merged from the main functiion

File: benchmarks/benchmark_serving.py [hunk 1]
```diff
@@ -185,6 +184,29 @@ def sample_sonnet_requests(
     return sampled_requests
 
 
+def sample_random_requests(input_len, output_len, num_prompts, range_ratio,
```
Comment (ywang96) [inline benchmarks/benchmark_serving.py:187]: Please add type hint for parameters in the function signature

File: benchmarks/benchmark_serving.py [hunk 2]
```diff
@@ -659,5 +691,20 @@ def main(args: argparse.Namespace):
         " format.",
     )
 
+    parser.add_argument("--random-input-len",
+                        type=int,
+                        default=1024,
+                        help="random sample input length")
```
Comment (ywang96) [inline benchmarks/benchmark_serving.py:697]:
```suggestion
                        help="Number of randomly sampled input tokens per request, used only for random dataset")
```

File: benchmarks/benchmark_serving.py [hunk 3]
```diff
@@ -659,5 +691,20 @@ def main(args: argparse.Namespace):
...
+
+    parser.add_argument("--random-output-len",
+                        type=int,
+                        default=128,
+                        help="random sample output length")
```
Comment (ywang96) [inline benchmarks/benchmark_serving.py:702]: Please update the help message per suggestion above.

File: benchmarks/benchmark_serving.py [hunk 4]
```diff
@@ -659,5 +691,20 @@ def main(args: argparse.Namespace):
...
+
+    parser.add_argument("--random-range-ratio",
+                        type=float,
+                        default=1.0,
+                        help="random sample range ratio")
```
Comment (ywang96) [inline benchmarks/benchmark_serving.py:707]: Please update the help message per suggestion above.

**Review by ywang96 — COMMENTED** [review]
Hey @haichuan1221! Sorry for the late review and thank you for the contribution!

Overall LGTM and I have left a few comments/suggestions

File: benchmarks/benchmark_serving.py (hunk 1 above)
Comment (haichuan1221) [reply benchmarks/benchmark_serving.py]: Just fixed, please check it again

Comment (ywang96) [reply benchmarks/benchmark_serving.py]: Update the formatting issues for you. For future reference, you can run the `format.sh` we provide in the repo for easily formatting your code changes.

**Review by ywang96 — APPROVED** [review]

File: benchmarks/benchmark_serving.py (hunk 1 above)
Comment (haichuan1221) [reply benchmarks/benchmark_serving.py]: OK, I will. Thinks for the tip

**Pull request merged**